lib/beaker-puppet/install_utils/puppet5.rb

```diff
@@ -0,0 +1,176 @@
+          artifact_filename = File.basename( artifact_url )
+          artifact_folder = File.dirname( artifact_url )
+          fetch_http_file( artifact_folder, artifact_filename, '.' )
+          onhost_package_dir = host.tmpdir( 'puppet_installer' )
+          scp_to host, artifact_filename, onhost_package_dir
+          onhost_package_file = "#{ onhost_package_dir }/#{ artifact_filename }"
+        elsif variant =~ /^(debian|ubuntu|cumulus)$/
+          onhost_package_file = host.tmpfile( 'puppet.deb' )
+          on host, "wget -O #{ onhost_package_file } #{ artifact_url }"
+        elsif variant == 'osx'
+          on host, "curl -O #{ artifact_url }"
+          onhost_package_file = "#{ project_name }*"
+        end
+
+        if variant == 'eos'
+          # TODO Will be refactored into {Beaker::Host#install_local_package}
+          #   immediately following this work. The release timing makes it
+          #   necessary to have this here separately for a short while
+          host.install_from_file( onhost_package_file )
+        elsif onhost_package_file
+          if onhost_package_dir
+            host.install_local_package( onhost_package_file, '.' )
+          else
+            host.install_local_package( onhost_package_file )
+          end
+        else
+          host.install_package( artifact_url )
+        end
+      end
+
+      # Sets up the repo_configs on the host for this build
+      #
+      # @param [Host] host Host to install repo_configs on
+      # @param [String] repoconfig_url URL to the repo_config
+      #
+      # @return nil
+      def install_repo_configs_on(host, repoconfig_url)
+        if repoconfig_url.nil?
+          logger.warn("No repo_config for host '#{host}'. Skipping repo_config install")
+          return
+        end
+
+        install_repo_configs_from_url( host, repoconfig_url )
+      end
+
+      # Installs a specified puppet project on all hosts. Gets build information
+      #   from the provided YAML file located at the +sha_yaml_url+ parameter.
+      #
+      # @param [String] sha_yaml_url URL to the <SHA>.yaml file containing the
+      #    build details
+      # @param [String] project_name Name of the project to install
+      #
+      # @note This install method only works for Puppet versions >= 5.0
+      #
+      # @return nil
+      def install_from_build_data_url(project_name, sha_yaml_url)
+        if !link_exists?( sha_yaml_url )
+          message = <<-EOF
+            <SHA>.yaml URL '#{ sha_yaml_url }' does not exist.
+            Please update the `sha_yaml_url` parameter to the `puppet5_install` method.
+          EOF
+          fail_test( message )
+        end
+
+        base_url, build_details = fetch_build_details( sha_yaml_url )
+        hosts.each do |host|
+          artifact_url, repoconfig_url = host_urls( host, build_details, base_url )
+          install_repo_configs_on( host, repoconfig_url )
+          install_artifact_on( host, artifact_url, project_name )
```

Comment:
This kind of looks like we're installing the repo_configs if they're available, but then not using those enabled repos? In `install_artifact_on`, it looks like we're just installing the artifact directly from the `artifact` url. Am I misinterpreting this?